Sources/Plasma/PubUtilLib/plGImage/plFont.cpp

```diff
@@ -1706,7 +1706,7 @@ class plBDFCharsParser : public plBDFSectParser
             {
                 // If we're doing data, all lines are hex values until we hit "ENDCHAR"
                 if( fRowsLeft == 0 )
-                    throw;
+                    throw false;
```

Comment:
Hmm, this is suspicious...  `throw` without any parameter will re-throw the current exception or call `std::terminate` if there isn't one...  But what does throwing a bool accomplish?  Who's catching from this call?  I feel like either changing this to throw a proper exception (like `std::runtime_error`) or an explicit crash (like `hsAssert(false)`) would be better.

Reply:
There are several other `throw` statements in this function that were throwing bools, so this was just making it consistent, but I'll try to investigate whether anything is actually catching these errors. Maybe `hsAssert` is a better option.

Reply:
So it appears that calls to parse the BDF contents are wrapped in a `try`/`catch` block, and catch both `std::exception` and non-exceptions: https://github.com/H-uru/Plasma/blob/5b47a7592770d2d1c96975266ac3d1e0b16e9bb3/Sources/Plasma/PubUtilLib/plGImage/plFont.cpp#L2100-L2112

It seems the only place this BDF code is actually used is in plFontConverter, and freetype2 has built-in BDF support so it's unclear to me if this code is still serving any purpose or if it should be removed.

Reply:
It looks like this should probably throw a real `std::exception` for logging purposes.